### DISCUSSION TITLE
Error! Binaries missing/corrupted.

### DISCUSSION THREAD

**Discussion by vojtafilipi** [jely2002/youtube-dl-gui#358] [General]
Hi, i just downloaded it and i got this error. I restarted PC and disable antivirus and it didn't helped to me. 😔
![Screenshot 2022-05-06 224824](https://user-images.githubusercontent.com/99132302/167214900-35dfd05f-ee7a-4315-829d-19018d7ee5f6.jpg)

**Comment by aremeszee** [comment]
I'm having the same problem

**Comment by nolanjurgens** [comment]
You can remove 2.4.0 and install 2.3.1 as a workaround until this gets fixed.

**Comment by Suvana** [comment]
Same issue for 2.3.1 as well.

**Comment by shadowofward** [comment]
Exact same issue. I thought it was because I was trying out Windows 11 but i guess not. If you uninstall 2.4.0 and install 2.3.1 it will work. However be careful to stop the near instant update that will occur the first time running 2.3.1. Uncheck Auto-update ytdl binary & Auto-update app. If it does download just select upgrade later. However although  2.3.1 does work it is super slow. I hope there is an update soon. This app is the best i have used to download any and all online vids. So thanks for the program and hope it starts working again soon :)

Well fixed it following some advice on another thread
https://github.com/jely2002/youtube-dl-gui/issues/354

the TL;DR
update back to 2.4.0
download yt-dlp.exe from 
https://github.com/yt-dlp/yt-dlp/releases
it's towards the bottom of the page
drop the yt-dlp.exe into this folder
C:\Users\***your personal user name blah blah blah***\AppData\Local\Programs\youtube-dl-gui\resources\app.asar.unpacked\binaries

and BANG! it works now and is extremely fast again! hope this helps

**Reply by vpreyer** [reply]
No, this doesn't work either. There isn't even a folder with that name. I guess there's no point in creating one if nothing refers to it. 
I'm on Windows 10, tried both portable and installer. No banana.

**Reply by mq003at** [reply]
> 

I am using the portable version, so when I first open it, it creates a folder called open-video-downloader, which stores all the components. You have to put yt-dlp.exe inside that folder.

[shadowofward](https://github.com/shadowofward) advice will work if you have the installer version.

**Reply by MotionMikeW** [reply]
Thanks for the help! works perfectly now.

**Reply by Knight2407** [reply]
Guys it really works. I just logged in, only to confirm. For those still trying let me simplify:
1. Install portable version from : open video github - [https://github.com/jely2002/youtube-dl-gui/releases](url)
2. Let it download in your default 'Download' directory
3. Install - and let the whole process complete
4. In downloads, a folder by the name - "open-video-downloader" will already be created
5. Download yt-dlp.exe from https://github.com/yt-dlp/yt-dlp/releases
6. Paste above file in "open-video-downloader"
7. Resart open video downloader and Bang it starts downloading metadata
8. Gratitude where its due @shadowofward and @mq003at - thanks both you guys! <3

**Reply by ackvf** [reply]
BTW, the portable exe is some kind of a self-executing zip. You can unzip it into it's own folder and run it from there for much faster loads.

**Reply by ashjaylk** [reply]
Thanks, this works for me

**Comment by SeanBeansShako** [comment]
i have the same problme pls fix it

**Reply by BenKLDEV** [reply]
Hi, I found the file in this location where YourUserName should be the username which you use on your PC.

C:\Users\YourUserName\AppData\Local\Programs\youtube-dl-gui\resources\app.asar.unpacked\binaries

and you paste it in this file and it should work. Make sure that the exe yt-dlp file, matches your operative system or it might not work.

Thank you for explaining it Shadowward :)

**Comment by ColinDavies71** [comment]
Had the same issue.... copying yt-dlp.exe into the folder as mentioned above resolves the problem (or if running the portable version into the 'open-video-downloader' folder it creates below itself)

The error code indeed is indicating that a binary file is indeed missing (or corrupt). And for some reason, yt-dlp.exe is no longer being installed when it's looking for the latest release.   Same for both the portable and the installer versions

Hope this helps.

**Comment by jely2002** [comment]
Heya! Thanks all for the workarounds, the app has since been rewritten in v3. Where this issue is fixed, and it should just work. Closing this as I'm cleaning up some old discussions.
